src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -2411,15 +2434,23 @@ emit_sri_vector (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *fsi
 				arg0_type = MONO_TYPE_I8;
 				cast_class = mono_defaults.int64_class;
 			}
-			arg_class = create_class_instance ("System.Runtime.Intrinsics", m_class_get_name (klass), m_class_get_byval_arg (cast_class));
+
+			const char *klass_name = m_class_get_name (klass);
+
+			if (!strcmp (klass_name, "Vector4")) {
```

Comment:
This can be either `Vector128` or `Vector4`, there are now other possible options right?

Reply:
right, at least until the issue blocking Vector2/3 handling is resolved.

Mono needs a more general helper for Vector2/3 to ensure the stores are created correctly; otherwise it asserts because the IR is in an invalid shape.